core/arch/arm/tee/tee_rpmb.c

```diff
@@ -413,9 +414,104 @@ static TEE_Result tee_rpmb_invoke(struct tee_rpmb_mem *mem)
 	return thread_rpc_cmd(TEE_RPC_RPMB_CMD, 2, params);
 }
 
+#ifdef CFG_ENC_FS
+static bool is_null_or_zero(const uint8_t *fek)
+{
+	int i;
+
+	if (!fek)
+		return true;
+	for (i = 0; i < TEE_FS_KM_FEK_SIZE; i++)
+		if (fek[i])
+			return false;
+	return true;
+}
+
+static TEE_Result crypt_block(uint8_t *out, const uint8_t *in,
+			      uint16_t blk_idx, const uint8_t *fek,
+			      const uint8_t *nonce, TEE_OperationMode mode)
+{
+	if (is_null_or_zero(fek)) {
+		/*
+		 * Unencrypted block, either because it does not belong to the
+		 * data area of a file (!fek), or because the file was created
+		 * with no encryption (e.g., by OP-TEE with CFG_ENC_FS=n).
+		 */
```

Comment:
Do we want to support a file created with a TEE not having CFG_ENC_FS, recompile the TEE with CFG_ENC_FS, and use the file?